MapboxCoreNavigation/Feedback.swift

```diff
@@ -3,89 +3,48 @@ import Foundation
 
-    /// Indicates an incorrect visual artifact.
-    case incorrectVisual
+    /// Indicates an incorrect visual.
+    case incorrectVisual(subtype: IncorrectVisualSubtype)
 
     /// Indicates confusing voice instruction.
-    case confusingAudio
+    case confusingAudio(subtype: ConfusingAudioSubtype)
 
     /// Indicates an issue with route quality.
-    case routeQuality
+    case routeQuality(subtype: RouteQualitySubtype)
 
     /// Indicates that an illegal route was recommended.
-    case illegalRoute
+    case illegalRoute(subtype: IllegalRouteSubtype)
 
     /// Indicates a road closure was observed.
-    case roadClosure
-
+    case roadClosure(subtype: RoadClosureSubtype)
     
     public var description: String {
         switch self {
         case .general:
             return "general"
-        case .incorrectVisual:
+        case .incorrectVisual(_):
             return "incorrect_visual"
-        case .confusingAudio:
+        case .confusingAudio(_):
             return "confusing_audio"
-        case .routeQuality:
+        case .routeQuality(_):
             return "route_quality"
-        case .illegalRoute:
+        case .illegalRoute(_):
             return "illegal_route"
-        case .roadClosure:
+        case .roadClosure(_):
             return "road_closure"
         }
     }
-
-    /// Returns a corresponding list of `FeedbackSubType`s for a given `FeedbackType`
-    public var subtypes: [FeedbackSubType] {
-        switch self {
-        case .general:
-            return []
-        case .incorrectVisual:
-            return [.turnIconIncorrect,
-                    .streetNameIncorrect,
-                    .instructionUnnecessary,
-                    .instructionMissing,
-                    .maneuverIncorrect,
-                    .exitInfoIncorrect,
-                    .laneGuidanceIncorrect,
-                    .roadKnownByDifferentName]
-        case .confusingAudio:
-            return [.guidanceTooEarly,
-                    .guidanceTooLate,
-                    .pronunciationIncorrect,
-                    .roadNameRepeated]
-        case .routeQuality:
-            return [.routeNonDrivable,
-                    .routeNotPreferred,
-                    .alternativeRouteNotExpected,
-                    .routeIncludedMissingRoads,
-                    .routeHadRoadsTooNarrowToPass]
-        case .illegalRoute:
-            return [.routedDownAOneWay,
-                    .turnWasNotAllowed,
-                    .carsNotAllowedOnStreet,
-                    .turnAtIntersectionUnprotected]
-        case .roadClosure:
-            return [.streetPermanentlyBlockedOff,
-                    .roadMissingFromMap]
-
-        }
-    }
-
-
-
 }
 
-/// Enum used to define the many `FeedbackSubType`s that may be nested under a given `FeedbackType`
-public enum FeedbackSubType: String {
-    /// Incorrect Visual Subtypes
+/// Enum denoting the subtypes of the  `Incorrect Visual` top-level category
```

Comment:
This way jazzy should automatically link the symbol:

```suggestion
/// Enum denoting the subtypes of `FeedbackType.incorrectVisual(subtype:)`.
```

(Ditto for the other enumerations.)